satellite/overlay/cache.go

```diff
@@ -69,6 +69,8 @@ type DB interface {
 	UpdateNodeInfo(ctx context.Context, node storj.NodeID, nodeInfo *pb.InfoResponse) (stats *NodeDossier, err error)
 	// UpdateUptime updates a single storagenode's uptime stats.
 	UpdateUptime(ctx context.Context, nodeID storj.NodeID, isUp bool, lambda, weight, uptimeDQ float64) (stats *NodeStats, err error)
+	// WithUpdateCache allows you to add an update cache to the DB
+	WithUpdateCache(updateCache *UpdateCache) DB
```

Comment:
Any reason to flip the dependency, such that UpdateCache calls into overlay.Cache rather than the other way around?

Reply:
Can you explain more? I'm not sure what you mean here.

Reply:
`UpdateCache` could also be written as a:

```
type CoalescedCache struct {
    overlay.Cache
    ...
}

func (cache *CoalescedCache) UpdateUptime(ctx context.Context ....) error {
```

Just wondering whether the resulting code would end up clearer.

Reply:
I can't telling if you're "wondering" in a Socratic sense and are trying to help me grasp what you already see (well, maybe more "rhetorical" than Socratic, as I'm sure Socrates would deny he actually saw anything), if you're genuinely curious of my opinion. :smile: 

Personally I think the current solution, after your other great suggestions, is fairly clean code-wise. But if you'd prefer I rework it I can do that.

Reply:
Currently, I'm really wondering :D. My gut feeling tells me that this would keep the overlay.Cache itself unmodified, and keeping the caching logic outside of the direct implementation there's less guessing in which path code takes. But there also might be some parts that I'm forgetting and it could end up more complicated.

Reply:
I went ahead and implemented this, thanks for the suggestion.